x-pack/legacy/plugins/lens/public/editor_frame_service/editor_frame/config_panel_wrapper.tsx

```diff
@@ -174,85 +189,365 @@ function LayerPanels(
-          <EuiButtonEmpty
-            size="xs"
-            iconType="trash"
-            color="danger"
-            data-test-subj="lns_layer_remove"
-            onClick={() => {
-              // If we don't blur the remove / clear button, it remains focused
-              // which is a strange UX in this case. e.target.blur doesn't work
-              // due to who knows what, but probably event re-writing. Additionally,
-              // activeElement does not have blur so, we need to do some casting + safeguards.
-              const el = (document.activeElement as unknown) as { blur: () => void };
+                          setState: (newState: unknown) => {
+                            props.updateVisualization(
+                              activeVisualization.setDimension({
+                                layerId,
+                                dimensionId: dimension.dimensionId,
+                                columnId: newId,
+                                prevState: props.visualizationState,
+                              })
+                            );
+                            props.updateDatasource(datasourceId, newState);
+                            setPopoverState({
+                              isOpen: true,
+                              openId: newId,
+                              addingToDimensionId: null, // clear now that dimension exists
+                            });
+                          },
+                        }}
+                      />
+                    )}
+                  </DragDrop>
+                ) : null}
+              </>
+            </EuiFormRow>
+          );
+        })}
 
-              if (el && el.blur) {
-                el.blur();
-              }
+        <EuiSpacer size="s" />
 
-              onRemove();
-            }}
-          >
-            {isOnlyLayer
-              ? i18n.translate('xpack.lens.resetLayer', {
-                  defaultMessage: 'Reset layer',
-                })
-              : i18n.translate('xpack.lens.deleteLayer', {
-                  defaultMessage: 'Delete layer',
-                })}
-          </EuiButtonEmpty>
-        </EuiFlexItem>
-      </EuiFlexGroup>
-    </EuiPanel>
+        <EuiFlexGroup justifyContent="center">
+          <EuiFlexItem grow={false}>
+            <EuiButtonEmpty
+              size="xs"
+              iconType="trash"
+              color="danger"
+              data-test-subj="lns_layer_remove"
+              onClick={() => {
+                // If we don't blur the remove / clear button, it remains focused
+                // which is a strange UX in this case. e.target.blur doesn't work
+                // due to who knows what, but probably event re-writing. Additionally,
+                // activeElement does not have blur so, we need to do some casting + safeguards.
+                const el = (document.activeElement as unknown) as { blur: () => void };
+
+                if (el && el.blur) {
```

Comment:
nitpick, we could probably do `el?.blur`

Reply:
Done